### DISCUSSION TITLE
Black Spots and Lines on Shadows

### DISCUSSION THREAD

**Discussion by jsnaxl** [bnpr/Malt#498] [Q&A]
I'm new to Malt and made a basic NPR shader with a gradient color ramp and used 2 suns in different lightgroups. for some reason these blackspots in the model start appearing and change its shape based on the view distance and angle. There are also lines appearing  in the shadow on the floor. I've also tried replacing the sun and making a new material and it's still there.

https://user-images.githubusercontent.com/91794266/236616774-f8419498-115e-47e6-9df6-220013028374.mp4

**Comment by pragma37** [comment]
Hi!
Those lines on the floor should be solved in the latest Development version.
I have no idea what could be causing the other issue, though. Could you upload the file?
